### ISSUE TITLE
Not capturing RF remotes

### ISSUE THREAD

**Issue by Tony-Coding** [rob121/broadlinkgo#2]
IR works great but RF not at all. It stays on 'Waiting for remote presses....' (red LED turns on on Broadlink) until I get the message 'Error: error while calling learn: learning timeout'
I have held the remote button continuously (I believe it continually transmits during this time as it has a transmit indicator and has long press functionality). Tried pulsing the remote button and also pressed the remote button before the button on the web interface.
I tried with another 433MHz remote as well.
In both cases, I had no issues learning RF commands using the Broadlink app IHC.

My device model is RM pro+. 
Your logs say Found a supported Broadlink RM3 Pro Plus, device type 10141 (0x279d)
'Broadlink Manager' another program I tried to use also won't capture RF and my device is identified as 'RM2 Pro Plus3' in that software.

**Comment by skarotech** [comment]
Hi, I have the same issue, it wont learn RF codes. I do see in your code there is support for RF, but I can't see any way to trigger this in the web gui ?

**Comment by rob121** [comment]
@Tony-Coding @skarotech  I don't have an RF version so I'm unable to test, if anyone has a spare I'm happy to do some testing.

**Comment by skarotech** [comment]
> @Tony-Coding @skarotech I don't have an RF version so I'm unable to test, if anyone has a spare I'm happy to do some testing.

Hi Rob, happy to send you one, just contact me directly info@skaro.com.au

**Comment by drphungky** [comment]
Just to add to the pile, I am having the same issue on a brand new RM pro+. I may end up returning it anyway because I didn't realize the RM Pro+ can't do a press and hold (which I need for my blinds.com blinds unfortunately), but figured I'd chime in before I do so, so you see it is not just you guys.

**Comment by rob121** [comment]
@skarotech thanks - I'm quite busy for the next month or two but I will reach out after!

**Comment by Tony-Coding** [comment]
@skarotech thanks for being able to supply one.
@rob121 I haven't found another solution so will wait till you can get around to this, thanks.

**Comment by D0m3L** [comment]
For your information guys - for me (Broadlink RM2, device type 10002 (0x2712)) learning and sending 433MHz codes works like a charm. 

```
Waiting for remote presses....
           

Code Detected!
2600e601090002690a00026207000265080002730700026909000271080002550700025e090002620b00025d0900026b0900026e0900025006140600026107000268090002dd0800026907000251070002660800024a080d09000264090002720800027809000275080002640900026809000287080002dd0800026f080002700c0002730800025d0800025c09000297060005340900046208000290070002580800027a09000265080002e10700026306000251080002710a00027f0800025f0800025608000264060c08000280080002850700025f0700028908000262080002680800029e07000285090002720600028308000273080002880900027e0700028b060007500700028207000262070002460700026806000257080004f70800028f090002710800026c0800028209000265070d08000269060002550800026f060c0800026e0800025809000284084a08000222090e09f80800024a09000277080002f1070002ca071005000284080003410800031a070002c4080002d70800028f070003110800025b080002640600025f080002bf0a00025d0900028d090002620800024e0bb5070002830800028b070002730700027908000267090002ac07000272080a090004ec070002700700027707000284080002680900026c060002530809070b07000d050251
Code Saved to /rob121/for/the/winner/cmd_testcode.txt
```

and the codes are from Sesoo light switch remotes (e.g: [this one](https://www.aliexpress.com/wholesale?catId=0&initiative_id=SB_20200229113505&SearchText=SESOO+Y601))

**Comment by dmellenthin** [comment]
@rob121 I would be more than happy to help if there's anything I can do.

Model: RM pro+ (on the sticker on the bottom)
PID: 0..0a9270000 (in the ihc Android App > Device Info)
I presume aligns with:
`knownDevice{deviceType: 0x27a9, name: "Broadlink RM3 Pro Plus v2", supported: true, ir: true, rf: true, power: false},`

IR works like a charm.  The light is on when learning.

RF (trying multiple 433Mhz remotes, which do Learn and function fine through the ihc Android App), result in:

    Waiting for remote presses....
           
    
    Error: error while calling learn: learning timeout

I would buy and ship you the same model as I have, but I'm not seeing it available for sale anymore (it looks like they've moved on to the RM4 pro).

**Comment by dmellenthin** [comment]
PROGRESS: https://github.com/rob121/broadlinkgo/pull/8

I didn't see `broadlink.LearnRF(...)` being called, so I added a checkbox (labeled **Is RF**) to the Learn page, and added a part to the call to `/learnchild` when checked.  I'm not sure if this is the perfect approach, but I'm no Go Pro, so for my first trick, I just did what I could to get it to learn, and sure as heck, it did Learn the RF command.

It seems to require a long-press (I wasn't able to learn a momentary button), and while it appears to send the command to my device, I have yet to confirm that it works with my device (It learns commands, and executing those commands indicates success, but my lamp isn't turning on.  BUT FWIW, I can't get learned buttons to work in the ifc Android App at this moment either; I remember it took some futzing and several tries in the past).

I think I may have invalidated the validation for a button name being specified [here](https://github.com/rob121/broadlinkgo/pull/8/files#diff-3002d695552ec7e7ebbaaf04a21d3439R293-R294).  So I wouldn't be offended if you want to touch that up :)

Great Work 👍  Thank You!!!  This is exactly what I was looking for and I can wait to play with this :)
